### DISCUSSION TITLE
Url validation for white spaces.

### DISCUSSION THREAD

**Discussion by andrea-cassioli-maersk** [pydantic/pydantic#5524] [Question]
I am using AnyHttpUrl to validate a uri coming from an Azure storage. The uri can include white spaces, but AnyHttpUrl raises an error. I see white spaces should not be used for security issues.

My question is whether there is a clear definition of the validation applied by AnyHttpUrl and the like.

**Comment by samuelcolvin** [comment]
V1 URL validation is not perfect, it should be better on V2.

Maybe you could try the V2 alpha, and let us know how you get on?

Given how complex URL validation is, I'm not keen to change it in v1 at this stage.

**Reply by andrea-cassioli-maersk** [reply]
Thanks @samuelcolvin. I can live with no whitespaces, it is more if there is a clear docs on what AnyHttpUrl and the like accept as valid URL.
